src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js

```diff
@@ -174,40 +192,55 @@ export default function CreateOfferRequest ({ history: { push } }) {
         name='notes'
         placeholder='What is this for?'
         ref={register} />
+
       <RecentCounterparties
         styleName='recent-counterparties'
         agents={agents}
         selectedAgentId={counterpartyId}
         selectAgent={selectAgent} />
-      <Button type='submit' wide variant='secondary' styleName='send-button' dataTestId='submit-button'>
-        {modeVerbs[mode]}
-      </Button>
+      <Button type='submit' wide variant='secondary' styleName='send-button' disabled={counterpartyId.length === AGENT_ID_LENGTH ? counterpartyNotFound : true}>Send</Button>
     </form>
   </PrimaryLayout>
 }
 
-export function RenderNickname ({ agentId, setCounterpartyNick }) {
-  const { loading, error, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
+export function RenderNickname ({ agentId, setCounterpartyNick, setErrorMessage, errorMessage, setCounterpartyNotFound, counterpartyNotFound, currentCounterpartyValue }) {
+  const { loading, error: queryError, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
     variables: { agentId }
   })
   const { nickname } = holofuelCounterparty
   useEffect(() => {
     setCounterpartyNick(nickname)
   }, [setCounterpartyNick, nickname])
 
+  if (!loading && holofuelCounterparty.notFound) {
+    errorMessage = 'This HoloFuel Peer is currently unable to be located in the network. \n Please verify the hash, ensure your HoloFuel Peer is online, and try again after a few minutes.'
+    counterpartyNotFound = true
+  } else if (!loading && !errorMessage && currentCounterpartyValue && currentCounterpartyValue.length === AGENT_ID_LENGTH) {
+    counterpartyNotFound = false
+  } else if (loading) counterpartyNotFound = true
```

Comment:
setting these variables (`errorMessage` and `counterpartyNotFound`) in the render method and then using them in `useEffect` is going to cause subtle bugs. I don't know if it's *technically* "creating side effects in the render method", but it causes the same problems.

Instead, all of this logic should be wrapped in `useEffect`. 

I think it's best if we take a look at this together.